### PR TITLE
Add upgrading to the latest version

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/pctl/issues/260

### TODO

- [x] Find a nice way to integration test this thing

### Checklist
- [x] Added tests that cover your change
- [x] Added/modified documentation as required (such as the `README.md`)
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: cmd/pctl/upgrade.go [hunk 1]
```diff
@@ -18,8 +19,12 @@ func upgradeCmd() *cli.Command {
 	return &cli.Command{
 		Name:      "upgrade",
 		Usage:     "upgrade profile installation",
-		UsageText: "To upgrade an installation: pctl upgrade pctl-profile-installation-path/ v0.1.1 ",
-		Flags:     createPRFlags,
+		UsageText: "To upgrade an installation: pctl upgrade pctl-profile-installation-path/ v0.1.1",
+		Flags: append(createPRFlags, &cli.BoolFlag{
+			Name:        "latest",
+			Usage:       "--latest",
+			DefaultText: "*WARNING*: Upgrade to the latest version.",
```
Comment (aclevername) [inline cmd/pctl/upgrade.go:26]:
```suggestion
			DefaultText: "*WARNING*: Upgrade to the latest version. Please ensure this is a valid upgrade path before proceeding",
```

File: cmd/pctl/upgrade.go [hunk 2]
```diff
@@ -36,12 +41,24 @@ func upgradeCmd() *cli.Command {
...
+		if c.Args().Len() != 1 {
+			return fmt.Errorf("please provid the path to the profile to upgrade to, e.g. pctl upgrade my-profile/")
+		}
+		profilePath = c.Args().Slice()[0]
+	} else {
+		if c.Args().Len() != 2 {
+			return fmt.Errorf("please provid the path to the profile and version to upgrade to, e.g. pctl upgrade my-profile/ v0.1.1")
+		}
+		profilePath = c.Args().Slice()[0]
+		profileVersion = c.Args().Slice()[1]
```
Comment (aclevername) [inline cmd/pctl/upgrade.go:59]: what about if `latest` was a valid value, rather than a flag :thinking: ?

Comment (Skarlso) [reply cmd/pctl/upgrade.go:59]: I don't really like that. If you see the previous push, I would have to then mock about with senver to compare the generated tags and values. Adds complexity. Also this is more explicit in my mind and more easy to discover as an option. While the <version> option doesn't make me think about , oh let's try writing `latest` as version to see what happens.

File: cmd/pctl/upgrade.go (hunk 1 above)
Comment (Skarlso) [reply cmd/pctl/upgrade.go:26]: I had `; whatever that is.` Yours is definitely more polite hahahahahahhaa.

**Review by aclevername — APPROVED** [review]

**Pull request merged**